### DISCUSSION TITLE
Thank you very much & Setup

### DISCUSSION THREAD

**Discussion by stevenhardy-digital** [diadal/universal-social-auth#2] [Q&A] [answered]
Hello @diadal 

I just want to say thank you very much for getting this new version of vue-social-auth live. 

Unfortunately being a Typescript novice, I am completely unsure how to set this up now within my Vue 2 SPA. Your README.md is good, but unfortunately, I am unable to follow it to be able to use this package within my application. 

Do you have any guides or example projects that I can take a look at please?

**Comment by diadal** [comment]
can you share your code with Codepen to i can see how to you config

**Comment by diadal** [comment]
you don't need to use typescript declaration if your app don't use that just import normal and use

**Comment by stevenhardy-digital** [comment]
Hello @diadal 

I have created a code pen to show you my set up. If you need access to the repo, then thats fine as well. [CodePen](https://codepen.io/stevenhardy-digital/pen/gOLJyEd)

**Reply by diadal** [reply]
https://diadal.github.io/universal-social-auth-test-vu2/ working check the example

**Reply by diadal** [reply]
did you get it work now with your project

**Reply by stevenhardy-digital** [reply]
I have for the ones that you have created :) I am just looking at the custom providers. By the way, I have supported you on Patron :) 

Sorry, can I be really cheeky and ask for an example of a custom config please? 

I see that I need these:

```
    scopeDelimiter: string;
    scopePrefix: string;
    name: string;
    state: string | Func;
    authorizationEndpoint: string;
    popupOptions: Record<string, unknown>;
    redirectUri: string;
    url: string;
    clientId: string | number;
    responseType: string;
    skipPooling: boolean;
```

**Reply by stevenhardy-digital** [reply]
```
const Twitch = {
  name: 'twitch',
  url: '/auth/twitch',
  authorizationEndpoint: 'https://id.twitch.tv/oauth2/authorize',
  defaultUrlParams: ['response_type', 'client_id', 'redirect_uri'],
  scope: 'user:edit%20user:read:email',
  scopeDelimiter: '%20',
  oauthType: '2.0',
  responseType: 'code',
  responseParams: {
    code: 'code'
  }
}
```
This is the same as your `MycustomProvider`

```
const options = {
  providers: {
    facebook: {
      clientId: '************',
      redirectUri: 'https://myapp.com/auth/facebook/callback'
    },
    twitch: {
      clientId: '*************',
      redirectUri: 'https://caniplaypc.test/auth/twitch/callback',
    }
  }
}
```

This redirects me to twitch, however it's not passing the scope through. So that comes back empty.

**Reply by stevenhardy-digital** [reply]
Hello @diadal 
Its nearly there! :) 

Just need to get the data to save in the laravel database and I have done it :) 

I have realised that in:
  `defaultUrlParams: ['response_type', 'client_id', 'redirect_uri'],`

I needed to add `'scope'`

**Reply by diadal** [reply]
let me `Twitch` add to test and to default

**Reply by stevenhardy-digital** [reply]
Thank you so much! Please let me know how you get on :)

**Reply by diadal** [reply]
working registering with twitch i have not account there

**Reply by diadal** [reply]
1: `composer require socialiteproviders/twitch`
2: https://socialiteproviders.com/Twitch/#installation-basic-usage
```
protected $listen = [
    \SocialiteProviders\Manager\SocialiteWasCalled::class => [
        // ... other providers
        'SocialiteProviders\\Twitch\\TwitchExtendSocialite@handle',
    ],
];

```

**Reply by diadal** [reply]
then you can get data like said in document

**Reply by diadal** [reply]
working ok

<img width="246" alt="Screen Shot 2021-03-19 at 9 19 52 PM" src="https://user-images.githubusercontent.com/19895516/111837924-e4fd3b00-88f8-11eb-9912-67ee79acd124.png">

**Reply by stevenhardy-digital** [reply]
Thank you so so much! I am working on that now. Thank you for all your help, I really appreciate it :)

**Reply by diadal** [reply]
let me you when you get it worked

**Reply by diadal** [reply]
working now i guess

**Reply by stevenhardy-digital** [reply]
Hello, sorry I haven’t had chance to look yet. I am sure it will work like you have said. :) I will let you know any problems

**Comment by diadal** [comment, marked as answer]
https://diadal.github.io/universal-social-auth-test-vu2/ working check the example
